### PR TITLE
Update http.client.rst

### PR DESCRIPTION
**Pull request by geryogam**
Fix some typos in the documentation of the [`http.client` standard library](https://docs.python.org/3/library/http.client.html).

### CODE REVIEW

File: Doc/library/http.client.rst [hunk 1]
```diff
@@ -572,7 +572,7 @@ Here is an example session that uses the ``HEAD`` method.  Note that the
    >>> data == b''
    True
 
-Here is an example session that shows how to ``POST`` requests::
+Here is an example session that uses the ``POST`` method::
```
Comment (merwok) [inline Doc/library/http.client.rst:581]: Why this change?

File: Doc/library/http.client.rst [hunk 2]
```diff
@@ -588,14 +588,13 @@ Here is an example session that shows how to ``POST`` requests::
...
+Client side ``PUT`` HTTP requests are very similar to ``POST`` HTTP requests. The
+difference lies only on the server side where HTTP servers will allow resources to
+be created via ``PUT`` requests. It should be noted that custom HTTP methods
 are also handled in :class:`urllib.request.Request` by setting the appropriate
-method attribute. Here is an example session that shows how to send a ``PUT``
-request using http.client::
+method attribute. Here is an example session that uses the ``PUT`` method::
 
-    >>> # This creates an HTTP message
+    >>> # This creates an HTTP request
```
Comment (merwok) [inline Doc/library/http.client.rst:603]: _request_ is the abstract concept, but _message_ is a type in this module, and used just below in the doc.
This line was not incorrect, I suggest reverting the change.

File: Doc/library/http.client.rst [hunk 3]
```diff
@@ -588,14 +588,13 @@ Here is an example session that shows how to ``POST`` requests::
    b'Redirecting to <a href="http://bugs.python.org/issue12524">http://bugs.python.org/issue12524</a>'
    >>> conn.close()
 
-Client side ``HTTP PUT`` requests are very similar to ``POST`` requests. The
-difference lies only the server side where HTTP server will allow resources to
-be created via ``PUT`` request. It should be noted that custom HTTP methods
+Client side ``PUT`` HTTP requests are very similar to ``POST`` HTTP requests. The
```
Comment (merwok) [inline Doc/library/http.client.rst]: Less clumsy in my opinion:
```suggestion
Client side HTTP ``PUT`` requests are very similar to ``POST`` requests. The
```

File: Doc/library/http.client.rst [hunk 4]
```diff
@@ -588,14 +588,13 @@ Here is an example session that shows how to ``POST`` requests::
...
+difference lies only on the server side where HTTP servers will allow resources to
+be created via ``PUT`` requests. It should be noted that custom HTTP methods
 are also handled in :class:`urllib.request.Request` by setting the appropriate
-method attribute. Here is an example session that shows how to send a ``PUT``
-request using http.client::
+method attribute. Here is an example session that uses the ``PUT`` method::
```
Comment (merwok) [inline Doc/library/http.client.rst:601]:
```suggestion
method attribute. Here is an example session that shows how to use the ``PUT`` method::
```

File: Doc/library/http.client.rst [hunk 5]
```diff
@@ -468,7 +468,7 @@ statement.
 
    Return the value of the header *name*, or *default* if there is no header
    matching *name*.  If there is more than one  header with the name *name*,
-   return all of the values joined by ', '.  If 'default' is any iterable other
+   return all of the values joined by ', '.  If *default* is any iterable other
```
Comment (merwok) [inline Doc/library/http.client.rst:477]: Thanks, these two changes are useful.

**Comment by slateny** [comment]
@maggyero Would you still be interested in addressing the comments above?

**Review by orsenthil — APPROVED** [review]
LGTM.

The only critical comment was reverting to `message` instead of `request`. but within the context, the usage of `request` was fine.

**Comment by merwok** [comment]
Some of my questions were left unanswered before this was merged, but it’s not very important.

For other PRs of the same style, please follow these guidelines:
- open an issue first, or a forum discussion
- propose substantive changes that do improve the text
- for cosmetic or debatable changes, get agreement on the issue first
- write a meaningful PR title
- address reviews

Thanks!

**Comment by orsenthil** [comment]
Hi @merwok ,

I made a note regarding your comment while reviewing it.

> The only critical comment was reverting to message instead of request. but within the context, the usage of request was fine.

Didn't want to keep the PR with other changes in open for a long time since your other comments were addressed. I acknowledge your other points.

Thank you.

**Comment by merwok** [comment]
Good, and no worry!
